core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -423,6 +425,23 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
         }
       }
     }
+    final OptionalServiceDependencies optionaAnnotation = clazz.getAnnotation(OptionalServiceDependencies.class);
+    if (optionaAnnotation != null) {
+      for (String className : optionaAnnotation.value()) {
+        try {
+          Class<?> dependency = ClassLoading.getDefaultClassLoader().loadClass(className);
+          if (Service.class.isAssignableFrom(dependency)) {
+            @SuppressWarnings("unchecked")
+            Class<? extends Service> serviceDependency = (Class<? extends Service>) dependency;
+            dependencies.add(serviceDependency);
+          } else {
+            throw new IllegalStateException("Service dependency declared by " + className + " is not a Service: " + dependency.getName());
+          }
+        } catch (ClassNotFoundException ignored) {
+          // dependency is optional so we ignore it
```

Comment:
This isn't the kind of optional I expected.  This kind of optional is:
> If this service interface is on the classpath then please use it, but if it's not on the classpath then please ignore it.

What I had expected was:
> If this service is in the dependency set for this environment already then please order my start/stop with respect to it. If it's not already in the set then don't pull it in on my account.